### PR TITLE
Fix for issue #47.

### PR DESCRIPTION
**Pull request by jlusiardi**
Introduces a new function in helper_lib.sh to query the command line
arguments of the running instances of a binary. This is done to get
rid of the problem of "-lf" versus "-alf" for pgrep.

### CODE REVIEW

**Comment by konstruktoid** [comment]
Thanks for the PR @jlusiardi, can you update to fix `tests/3_docker_daemon_configuration_files.sh` as well? See https://github.com/docker/docker-bench-security/pull/46#issuecomment-115108878.

**Comment by jlusiardi** [comment]
@konstruktoid of course I can, sorry that I've missed it in the first place. My grep foo was weak ;)

**Comment by konstruktoid** [comment]
Thanks @jlusiardi, looks good. I'll run the PR on a couple of distros later on and most likely merge after that.

**Comment by konstruktoid** [comment]
Works when running in a container on Ubuntu, Fedora and CoreOS, but fails when running in the shell on Fedora:

```sh
ESC[1;34m[INFO]ESC[0m 2 - Docker Daemon Configuration
cat: /proc/28954/cmdline: No such file or directory
ESC[1;32m[PASS]ESC[0m 2.1  - Do not use lxc execution driver
cat: /proc/28968/cmdline: No such file or directory
ESC[1;32m[PASS]ESC[0m 2.2  - Restrict network traffic between containers
cat: /proc/28982/cmdline: No such file or directory
ESC[1;32m[PASS]ESC[0m 2.3  - Set the logging level
cat: /proc/28996/cmdline: No such file or directory
ESC[1;32m[PASS]ESC[0m 2.4  - Allow Docker to make changes to iptables
cat: /proc/29010/cmdline: No such file or directory
```

```sh
$ cat /etc/redhat-release
Fedora release 22 (Twenty Two)
```

```sh
$ cat /etc/lsb-release
DISTRIB_ID=CoreOS
DISTRIB_RELEASE=681.2.0
DISTRIB_CODENAME="Red Dog"
DISTRIB_DESCRIPTION="CoreOS 681.2.0"
```

```sh
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=15.04
DISTRIB_CODENAME=vivid
DISTRIB_DESCRIPTION="Ubuntu 15.04"
```

**Comment by jlusiardi** [comment]
@konstruktoid can you give me more information on how you ran into the error? 

At least a cmdline file seems to exist on Fedora 22 (which was my first guess for the error):
```
root@newbox ~ # docker run -t -i --rm fedora /bin/bash
[root@3272ac047300 /]# sleep 5m&
[1] 19
[root@3272ac047300 /]# ls /proc/19/cmdline 
/proc/19/cmdline
[root@3272ac047300 /]# cat /proc/19/cmdline 
sleep5m[root@3272ac047300 /]# cat /etc/redhat-release 
Fedora release 22 (Twenty Two)
[root@3272ac047300 /]# 
```

Any hints are welcome!

**Comment by konstruktoid** [comment]
@jlusiardi the code runs well when you build and run the container. But running the script directly on a Fedora host, the basic `sudo sh ./docker-bench-security.sh`, generates the errors.

File: helper_lib.sh [hunk 1]
```diff
@@ -36,3 +36,13 @@ contains() {
         return 1    # $substring is not in $string
     fi
 }
+
+# Extracts all commandline args from all running processes named like the first parameter
+get_command_line_args() {
+    PROC="$1"
+
+    for PID in `pgrep $PROC`
```
Comment (rnelson0) [inline helper_lib.sh]: This should probably use `pgrep -x $PROC`

File: tests/2_docker_daemon_configuration.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ info "2 - Docker Daemon Configuration"
 
 # 2.1
 check_2_1="2.1  - Do not use lxc execution driver"
-pgrep -lf docker | grep lxc >/dev/null 2>&1
+get_command_line_args docker | grep lxc >/dev/null 2>&1
```
Comment (rnelson0) [inline tests/2_docker_daemon_configuration.sh]: As these updated checks will look for the pattern (i.e. `lxc`) on a single line, can the pattern be improved to match the entire line, or as much of the line is reasonable? i.e. `^lxc`, `lxc$`, etc. Similar changes for each of the patterns below as well.

**Comment by jlusiardi** [comment]
@konstruktoid i think the issue come from short lived docker processes (most likely client calls) in parallel to running the script. 

```
docker run -t -i --rm debian sleep 5m&
pgrep -x docker
```
yields 2 PIDs (daemon and client)

IMHO, client calls should be ignored completely. Can we use `prep -x -o` to get only the oldest (the daemon) docker calls?

@rnelson0  thanks for your hint!

**Comment by konstruktoid** [comment]
Thanks!
OK on Fedore, CoreOS and Ubuntu.

**Comment by jlusiardi** [comment]
great thanks!

**Pull request merged**